### PR TITLE
Prepare the plugin for CakePHP 3.6

### PR DESCRIPTION
**Pull request by HavokInspiration**
With CakePHP 3.6 just around the corner and all the deprecation notices it throws (and the fact I need it for an app :D), this PR aims to prepare it for 3.6.

I'm targeting master but I can change for another branch if you want to keep compatibility for versions < 3.4.0.
I can also make changes to the `composer.json`.

Just tell me however the way you want to manage this new gap.

### CODE REVIEW

**Comment by HavokInspiration** [comment]
There is probably a need to update the Travis matrix as well since it should at least target 3.4.0 now given the changes and when the new methods were introduced.

**Comment by HavokInspiration** [comment]
Well actually, it would for 3.6.0 only since some methods linked to associations were introduced in 3.6.0 (`getAssociation()` / `hasAssociation()`)

**Comment by HavokInspiration** [comment]
PR updated : I locked the cakephp dependencies to `3.6.0-beta1` and removed PHP 5.4 and 5.5 from the travis matrix.

File: src/Model/Behavior/TrashBehavior.php [hunk 1]
```diff
@@ -320,7 +320,7 @@ public function getTrashField($aliased = true)
                 throw new RuntimeException('TrashBehavior: "field" config needs to be provided.');
             }
 
-            $this->config('field', $field);
+            $this->getConfig('field', $field);
```
Comment (ADmad) [inline src/Model/Behavior/TrashBehavior.php]: This should be `setConfig()`, seems this line doesn't have test coverage.

Comment (HavokInspiration) [reply src/Model/Behavior/TrashBehavior.php]: I'll take a look

**Comment by HavokInspiration** [comment]
@ADmad I took care of your comment and added a test to cover that line.

File: .travis.yml [hunk 2]
```diff
@@ -1,8 +1,6 @@
 language: php
 
 php:
-  - 5.4
-  - 5.5
   - 5.6
   - 7.0
   - 7.1
```
Comment (ADmad) [inline .travis.yml]: Can you please add 7.2 also to the matrix?

**Comment by HavokInspiration** [comment]
PHP 7.2 added to the matrix.

**Comment by ADmad** [comment]
Thanks!

**Pull request merged**